pkg/ansible/watches/watches.go

```diff
@@ -136,10 +136,53 @@ func (w *Watch) UnmarshalYAML(unmarshal func(interface{}) error) error {
 	w.WatchClusterScopedResources = tmp.WatchClusterScopedResources
 	w.Finalizer = tmp.Finalizer
 	w.AnsibleVerbosity = getAnsibleVerbosity(gvk, ansibleVerbosityDefault)
+	w.addRolePlaybookPaths()
 
 	return nil
 }
 
+// addRolePlaybookPaths will add the full path for rules an playbooks
+// when the relative one is informed. Note that these files are moved in the build
+// to /opt/ansible always
+func (w *Watch) addRolePlaybookPaths() {
+	w.Playbook = getFullPath(w.Playbook)
+	w.Role = getFullPath(w.Role)
+	if w.Finalizer != nil && len(w.Finalizer.Role) > 0 {
+		w.Finalizer.Role = getFullPath(w.Finalizer.Role)
+	}
+	if w.Finalizer != nil && len(w.Finalizer.Playbook) > 0 {
+		w.Finalizer.Playbook = getFullPath(w.Finalizer.Playbook)
+	}
+}
+
+// getFullPath will return a valid full path for the role and/or playbook
+func getFullPath(path string) string {
+	if isPathRelative(path) {
+		if k8sutil.IsRunModeLocal() {
+			return addCurrentDirPath(path)
+		}
+		return addDefautImagePath(path)
+	}
+	return path
+}
+
+// isPathRelative returns true when the path informed is relative
+func isPathRelative(path string) bool {
+	return len(path) > 0 && !strings.HasPrefix(path, "/")
+}
+
+// addDefautImagePath will concat the relative path with the dir path were
+// the roles and playbooks are copied in the images
+func addDefautImagePath(path string) string {
+	return "/opt/ansible/" + path
+}
+
+// addCurrentDirPath will concat the relative path with the current directory path
+// Note: just when the operator is executed out of the cluster it is valid
+func addCurrentDirPath(path string) string {
+	return projutil.MustGetwd() + "/" + path
```

Comment:
No need for a separate function for this. Instead, use `filepath.Join(projutil.MustGetwd(), path)`, which is cross-platform. Not all OSes use `/` as the filepath separator.